boolean.js

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+'use strict';
+
+var bufrw = require('bufrw');
+var expected = require('bufrw/errors').expected;
+
+var BooleanRW = bufrw.Base(
+    bufrw.UInt8.byteLength,
+    readTBooleanFrom,
+    writeTBooleanInto);
+
+function readTBooleanFrom(buffer, offset) {
+    var res = bufrw.UInt8.readFrom(buffer, offset);
+    if (!res.err) {
+        res.value = Boolean(res.value);
+    }
+    return res;
```

Comment:
It is a "prevailing pattern" except when it's not, like here:
- in tail of an R/L/W method
- where both the error case and success case want to just `return res`
- because the success case wants to mutate / re-use the already-allocated res